ydb/core/blobstorage/dsproxy/dsproxy_get_impl.cpp

```diff
@@ -391,4 +466,103 @@ void TGetImpl::OnVPutResult(TLogContext &logCtx, TEvBlobStorage::TEvVPutResult &
+        if (item.Data != partToCheck) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool TGetImpl::FindCorruptedPart42(const TBlobState &blobState, const TRope &data, ui32 &outPartIndex) {
+    const auto &parts = blobState.PartMap;
+    auto crcMode = static_cast<TErasureType::ECrcMode>(blobState.Id.CrcMode());
+    TStackVec<TRope, TypicalPartsInBlob> selectedParts(Info->Type.TotalPartCount());
+    TStackVec<TRope, TypicalPartsInBlob> restoredParts(Info->Type.TotalPartCount());
+    TRope restoredBlob;
+
+    for (ui32 excludedPart = 0; excludedPart < parts.size(); ++excludedPart) {
+        if (parts[excludedPart].Data.IsEmpty()) {
+            continue;
+        }
+        std::fill(selectedParts.begin(), selectedParts.end(), TRope());
+        std::fill(restoredParts.begin(), restoredParts.end(), TRope());
+        ui32 restoreMask = 0;
+        ui32 selectedCount = 0;
+        for (ui32 i = 0; i < parts.size(); ++i) {
+            if (i == excludedPart || parts[i].Data.IsEmpty()) {
+                continue;
+            }
+            ui32 partId = parts[i].PartIdRequested - 1;
+            if (selectedParts[partId].IsEmpty()) {
+                selectedParts[partId] = parts[i].Data;
+                restoreMask |= (1 << partId);
+                ++selectedCount;
+            }
+            if (selectedCount == 4) {
+                break;
+            }
+        }
+        if (selectedCount != 4) {
+            continue;
+        }
+        
+        restoredBlob.clear();
+        ErasureRestore(crcMode, Info->Type, data.size(), &restoredBlob, selectedParts, restoreMask);
+        ErasureSplit(crcMode, Info->Type, restoredBlob, restoredParts);
+        
+        ui32 mismatch = 0;
+        for (ui32 i = 0; i < parts.size(); ++i) {
+            if (i == excludedPart || parts[i].Data.IsEmpty()) {
+                continue;
+            }
+            ui32 partId = parts[i].PartIdRequested - 1;
+            if (parts[i].Data != restoredParts[partId]) {
+                ++mismatch;
+            }
+        }
+        if (mismatch == 0) {
+            outPartIndex = excludedPart;
+            return true;
+        }
+    }
+    return false;
+}
+
+bool TGetImpl::FindCorruptedPartMirror(const TBlobState &blobState, ui32 &outPartIndex) {
+    const auto &parts = blobState.PartMap;
+    ui32 corruptedParts = 0;
+
+    for (ui32 checkedPart = 0; checkedPart < parts.size(); ++checkedPart) {
+        if (parts[checkedPart].Data.IsEmpty()) {
+            continue;
+        }
+        ui32 mismatch = 0;
+        for (ui32 i = 0; i < parts.size(); ++i) {
+            if (i == checkedPart || parts[i].Data.IsEmpty()) {
+                continue;
+            }
+            if (parts[checkedPart].Data != parts[i].Data) {
+                ++mismatch;
+            }
+        }
+        if (mismatch > 1) {
+            outPartIndex = checkedPart;
+            ++corruptedParts;
+        }
+    }
+    return corruptedParts == 1;
```

Comment:
what if there are more than one corruptedParts? The function will return false, indicating that everything is consistent?